src/include/OpenImageIO/fmath.h

```diff
@@ -458,11 +460,15 @@ ifloor (float x)
 /// to the built-in modf, but returns a true int, always rounds down
 /// (compared to modf which rounds toward 0), and always returns
 /// frac >= 0 (comapred to modf which can return <0 if x<0).
-inline float
+inline OIIO_HOSTDEVICE float
 floorfrac (float x, int *xint)
 {
 #if 1
+#ifndef __CUDACC__
     float f = std::floor(x);
```

Comment:
Just curious...

If instead of the whole `#ifndef __CUDACC__` construct, we just had

    inline OIIO_HOSTDEVICE float fast_blah (
        using std::blah;  // or even:  using namespace std
        float f = blah(x);
        ...

would that do the right thing on both/all platforms? That is, would that idiom use std::blah if it exists, and an unqualified `blah` if it didn't? If that works, I think it may be a less cluttered approach for the functions where the only Cuda/C++ difference is prefixing with std::.  Or do I misunderstand how the namespace resolution would work?

Reply:
Good question. I'll investigate this and get back to you.

Reply:
Something I may not have considered here:  if the Cuda compiler sees a std::floor and also an unqualified floor, and you are purposely selecting the unqualified one on Cuda (is there some reason to not want std?), then I think the `#if` would be necessary. But if Cuda simply lacks the std namespace altogether, then I think my `using` suggestion just might do the right thing.

Reply:
Did that work? Or should I commit as-is?

Reply:
It did work for the most part. ```std::min``` seems to be a holdout seemingly based on a template vs. overloading issue. I will push an updated version shortly.

Reply:
Don't over-strain yourself to make the idiom fit where it's difficult. Do it where it simplifies the code easily, just leave what you have to where it doesn't.